environment/src/main/java/com/sequenceiq/environment/environment/dto/AuthenticationDtoConverter.java

```diff
@@ -21,12 +21,9 @@ public AuthenticationDtoConverter(PublicKeyValidator publicKeyValidator) {
 
     public EnvironmentAuthentication dtoToAuthentication(AuthenticationDto authenticationDto) {
         EnvironmentAuthentication environmentAuthentication = new EnvironmentAuthentication();
-        if (StringUtils.isNotEmpty(authenticationDto.getPublicKey())) {
-            ValidationResult validationResult = publicKeyValidator.validatePublicKey(authenticationDto.getPublicKey());
-            if (!validationResult.hasError()) {
-                List<String> parts = Arrays.asList(StringUtils.split(authenticationDto.getPublicKey(), " "));
-                environmentAuthentication.setPublicKey(String.format("%s %s %s", parts.get(0), parts.get(1), authenticationDto.getLoginUserName()));
-            }
+        if (isValidSshKey(authenticationDto.getPublicKey())) {
```

Comment:
dont we need to check first that the public key exists?

Reply:
it is the raw public key. We want to validate the format here. However, I moved the conditions onto a method. The conditions are the same than previously.